dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultAuthenticationService.java

```diff
@@ -41,19 +45,24 @@ public class DefaultAuthenticationService implements AuthenticationService {
 
   @Override
   public void obtainAuthentication(String userId) throws NotFoundException {
-    if (userId == null) {
-      clearAuthentication();
-      return;
+    Object username = "System";
+    Collection<? extends GrantedAuthority> authorities = List.of(new SimpleGrantedAuthority("ALL"));
```

Comment:
It feels bad mixing these 2 scenarios like this (valid userId v system call). At best it works as expected. At worst, something happens with the `userId` and the current user has full access it seems, albeit with a different username and no password.

Reply:
Yes, I don't like this either. Any better ideas?